frontend/public/components/monitoring/receiver-forms/alert-manager-receiver-forms.tsx

```diff
@@ -222,12 +217,9 @@ const ReceiverBaseForm: React.FC<ReceiverBaseFormProps> = ({
   editReceiverNamed,
   alertmanagerGlobals, // contains default props not in alertmanager.yaml's config.global
 }) => {
-  const [errorMsg, setErrorMsg] = React.useState<string>();
+  const { config, errorMessage: loadErrorMsg } = getAlertmanagerConfig(secret);
+  const [errorMsg, setErrorMsg] = React.useState<string>(loadErrorMsg);
```

Comment:
Looks like we only care about the result of `getAlertmanagerConfig` the first time because `loadErrorMsg` is set in the state and a subsequent render can never result in a new error message in the state.

This differs from the original impl which could call `setErrorMsg` on subsequent renders.

Reply:
I updated my PR now which includes also some small fixes for the form when the `getAlertmanagerConfig` returns an error. I saw this different scenarios and tested them on the master branch and with this PR.

You need to update the alertmanager-main secret to reproduce some of these errors.

(console path `/k8s/ns/openshift-monitoring/secrets/alertmanager-main/yaml`)

1. YAML is in a good state: Works fine before and with this PR.

2. There is no `data.alertmanager.yaml`: The complete Alertmanager shows an error (before and with this PR):

![image](https://user-images.githubusercontent.com/139310/119190349-83cf2080-ba7d-11eb-995b-86604e616571.png)

3. The YAML contains invalid or incompleted data, like a string or an object without missing receivers, etc.:

secret (with base64 encoded yaml `dafa` - a string):
```
data:
  alertmanager.yaml: ZGFmYQ==
```

secret (with base64 encoded yaml `dafa:` - an object):
```yaml
data:
  alertmanager.yaml: ZGFmYToK
```

Before: Crash because some expected properties are not defined in the config.

After: Because the most code use lodash or some other fallbacks (like ` || {}`) I added these fallbacks to the missing senarios. The form shows now an empty, not configured form.


Finally: The loading and patch error message was just shown to the user. The save button was not disabled. Instead of setting the loadError as state variable, I just show the patch or loading message, and prefers the patch message.

Reply:
The new changes look much better. 
Handling the two error states separately is also cleaner solution.

Reply:
/cc @kyoto